semver/mod.ts

```diff
@@ -231,1720 +230,91 @@
-  options?: Options,
-): 1 | 0 | -1 {
-  const versionA = new SemVer(a, options);
-  const versionB = new SemVer(b, options);
-  return versionA.compare(versionB) || versionA.compareBuild(versionB);
-}
-
-/**
- * Returns `0` if `v1 == v2`, or `-1` if `v1` is greater, or `1` if `v2` is
- * greater. Sorts in descending order if passed to `Array.sort()`,
- */
-export function rcompare(
-  v1: string | SemVer,
-  v2: string | SemVer,
-  options?: Options,
-): 1 | 0 | -1 {
-  return compare(v2, v1, options);
-}
-
-export function sort<T extends string | SemVer>(
-  list: T[],
-  options?: Options,
-): T[] {
-  return list.sort((a, b) => {
-    return compareBuild(a, b, options);
-  });
-}
-
-export function rsort<T extends string | SemVer>(
-  list: T[],
-  options?: Options,
-): T[] {
-  return list.sort((a, b) => {
-    return compareBuild(b, a, options);
-  });
-}
-
-/** Greater than comparison */
-export function gt(
-  v1: string | SemVer,
-  v2: string | SemVer,
-  options?: Options,
-): boolean {
-  return compare(v1, v2, options) > 0;
-}
-
-/** Less than comparison */
-export function lt(
-  v1: string | SemVer,
-  v2: string | SemVer,
-  options?: Options,
-): boolean {
-  return compare(v1, v2, options) < 0;
-}
-
-/**
- * This is true if they're logically equivalent, even if they're not the exact
- * same string.
- */
-export function eq(
-  v1: string | SemVer,
-  v2: string | SemVer,
-  options?: Options,
-): boolean {
-  return compare(v1, v2, options) === 0;
-}
+import {
+  comparatorMax,
+  comparatorMin,
+  SemVerComparator,
+} from "./comparator.ts";
+import { tryParse, tryParseComparator, tryParseRange } from "./parse.ts";
+import { rangeTest, SemVerRange } from "./range.ts";
+import { SemVer } from "./semver.ts";
+import { isSemVer, isSemVerComparator, isSemVerRange } from "./validity.ts";
+
+export * from "./comparator.ts";
+export * from "./format.ts";
+export * from "./operators/mod.ts";
+export * from "./parse.ts";
+export * from "./range.ts";
+export * from "./semver.ts";
+export * from "./types.ts";
 
 /**
- * This is true if they're not logically equivalent, even if they're the exact
- * same string.
+ * This just wraps tryParse with the intent to be compatible with an older version of this api.
+ * @param value A string which may or may not contain a valid SemVer
+ * @returns The original value if valid or undefined
  */
-export function neq(
-  v1: string | SemVer,
-  v2: string | SemVer,
-  options?: Options,
-): boolean {
-  return compare(v1, v2, options) !== 0;
-}
-
-/** Greater than or equal comparison */
-export function gte(
-  v1: string | SemVer,
-  v2: string | SemVer,
-  options?: Options,
-): boolean {
-  return compare(v1, v2, options) >= 0;
-}
-
-/** Less than or equal comparison */
-export function lte(
-  v1: string | SemVer,
-  v2: string | SemVer,
-  options?: Options,
-): boolean {
-  return compare(v1, v2, options) <= 0;
+export function valid(value: string | undefined): string | undefined {
```

Comment:
Not sure if we should provide this function in the long run. I think the user should use a try-catch block on parse if he wants to check. Maybe add a deprecation note?